simplecpp.cpp

```diff
@@ -1282,12 +1270,99 @@ void simplecpp::TokenList::constFoldBitwise(Token *tok)
                 result = (stringToLL(tok->previous->str()) ^ stringToLL(tok->next->str()));
             else /*if (*op == '|')*/
                 result = (stringToLL(tok->previous->str()) | stringToLL(tok->next->str()));
-            tok = tok->previous;
-            tok->setstr(toString(result));
-            deleteToken(tok->next);
-            deleteToken(tok->next);
+            simpleSquash(tok, toString(result));
+        }
+    }
+}
+
+static simplecpp::Token * stepForward(simplecpp::Token *tok)
+{
+    if (!tok)
+        return nullptr;
+    return tok->next;
+}
+
+static simplecpp::Token * stepBack(simplecpp::Token *tok)
+{
+    if (!tok)
+        return nullptr;
+    return tok->previous;
+}
+
+static int getTokensDeleteCount(simplecpp::Token *tok, const std::set<std::string> & breakPoints, simplecpp::Token *(step)(simplecpp::Token *), const std::pair<std::string, std::string> & brackets)
```

Comment:
hmm I am not sure about the step function pointer . It could be a `bool forward` flag instead. that provides less freedom.